### PR TITLE
setup: add cross compiling machine

### PR DESCRIPTION
**Pull request by joaocgreis**
Currently supports Raspberry Pi 1. Based on Ubuntu 14.04.

### CODE REVIEW

**Comment by joaocgreis** [comment]
- [ ] `grep -q ^github.com ~/.ssh/known_hosts || (ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts)`

**Comment by joaocgreis** [comment]
@nodejs/build any objections to just landing this? This has been working well enough for 2 years and should be in master. There can probably be improvements, but those can be done in other PRs.

**Comment by gibfahn** [comment]
Let's just land it

File: setup/cross-compiler/ansible-playbook.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      apt: update_cache=yes
+      tags: general
+
+    - name: General | APT Upgrade
+      apt: upgrade=full
+      tags: general
+
+    - name: General | Install required packages
+      apt: name={{ item }} update_cache=yes state=latest
+      with_items: packages
```
Comment (seishun) [inline setup/cross-compiler/ansible-playbook.yaml]: @joaocgreis This gives me the following error:

```
failed: [nodejs-cross-compiler] (item=[u'packages']) => {"changed": false, "item": ["packages"], "msg": "No package matching 'packages' is available"}
```

File: setup/cross-compiler/ansible-playbook.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    - name: Raspberry | Copy cc-armv6.sh script
+      copy: src=./resources/cc-armv6.sh dest=/opt/raspberrypi/cc-armv6.sh mode=0755
+      tags: raspberry
+
+    - name: Raspberry | Copy cc-armv7.sh script
+      copy: src=./resources/cc-armv7.sh dest=/opt/raspberrypi/cc-armv7.sh mode=0755
+      tags: raspberry
+
+    - name: User | Download pubkey(s)
+      get_url: url=https://github.com/{{ item }}.keys dest=/tmp/{{ item }}.keys
```
Comment (seishun) [inline setup/cross-compiler/ansible-playbook.yaml]: This results in:

```
failed: [nodejs-cross-compiler -> 127.0.0.1] (item=ssh_users) => {"changed": false, "dest": "/tmp/ssh_users.keys", "item": "ssh_users", "msg": "Request failed", "response": "HTTP Error 404: Not Found", "state": "absent", "status_code": 404, "url": "https://github.com/ssh_users.keys"}
```

File: setup/cross-compiler/ansible-playbook.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      delegate_to: 127.0.0.1
+      become: no
+      with_items: ssh_users
+      tags: user
+
+    - name: General | Create authorized_keys for root
+      authorized_key: user="root" key="{{ lookup('file', '/tmp/' + item + '.keys') }}"
+      with_items: ssh_users
```
Comment (seishun) [inline setup/cross-compiler/ansible-playbook.yaml]: Ditto.

**Comment by joaocgreis** [comment]
@seishun all of those seem to have the same root cause. These scripts were developed for Ansible 1, and worked well at the time. Ansible 2 seems to have changed things: https://github.com/ansible/ansible/commit/2e39661a26d881f1ff5991ae46e5cbf45b91cfe9

The variables are defined in `ansible-vars.yaml`. The solution seems simple, just change `with_items: packages` to `with_items: "{{packages}}"`. The steps for ssh keys can be removed because we now use a shared key (that is already on the machine).

**Pull request merged**